### PR TITLE
acceptance: PUT /users/<id>

### PR DESCRIPTION
**Pull request by mchalski**
Issue: https://tracker.mender.io/browse/MEN-1285

also contains some bugfixes for the service - one critical (user was never updated in the db if MT was enabled).

@maciejmrowiec @mendersoftware/rndity

### CODE REVIEW

File: tests/tests/client.py [hunk 1]
```diff
@@ -90,6 +90,16 @@ def create_user(self, user, auth=None):
...
 
+    def update_user(self, uid, update, auth=None):
+        if auth is None:
+            auth=self.auth
+
+        return self.client.users.put_users_id(id=uid, user_update=update, _request_options={"headers": auth}).result()
+
+    def login(self, username, password):
+        auth = common.make_basic_auth(username, password)
+        return self.client.auth.post_auth_login(Authorization = auth).result()
```
Comment (bboozzoo) [inline tests/tests/client.py]: `Authorization=auth`

File: tests/tests/test_api_management.py [hunk 2]
```diff
@@ -178,3 +178,153 @@ def test_ok(self, tenant_id, api_client_mgmt, init_users_mt):
...
+        assert len(users) == len(init_users)
+
+        found = [u for u in users if u.email == user.email]
+        assert len(found) == 1
+
+        # try if login still works
+        _, r = api_client_mgmt.login(user.email, update["password"])
+        assert r.status_code == 200
+
+    def _do_test_ok_email_and_pass(self, api_client_mgmt, init_users, user, update, tenant_id=None):
```
Comment (bboozzoo) [inline tests/tests/test_api_management.py]: isn't this a copy of `_do_test_ok_pass` ?

File: tests/tests/test_api_management.py [hunk 3]
```diff
@@ -178,3 +178,153 @@ def test_ok(self, tenant_id, api_client_mgmt, init_users_mt):
...
+
+    def _do_test_fail_not_found(self, api_client_mgmt, init_users, update, tenant_id=None):
+        auth = None
+        if tenant_id is not None:
+            auth = make_auth("foo", tenant_id)
+
+        try:
+            _, r = api_client_mgmt.update_user("madeupid", update, auth)
+        except bravado.exception.HTTPError as e:
+            assert e.response.status_code == 404
```
Comment (kjaskiewiczz) [inline tests/tests/test_api_management.py]: test will pass even if the device with "madeupid" will be found, is this ok?

Comment (mchalski) [inline tests/tests/test_api_management.py]: yup, we control the init data set and 'madeupid' is dumb enough that we won't deliberately insert it to the db

File: tests/tests/test_api_management.py (hunk 2 above)
Comment (mchalski) [inline tests/tests/test_api_management.py]: refactored

File: tests/tests/client.py (hunk 1 above)
Comment (mchalski) [inline tests/tests/client.py]: fixed

**Pull request merged**